src/API.jl

```diff
@@ -0,0 +1,557 @@
+results in a global dictionary, and batching larger requests with this function
+similarly increases performance.
+"""
+function json(name::AbstractString)
+    return json([name])[normalize_name(name)]
+end
+
+"""
+json(pkg::BrewPkg)
+
+Return the full JSON object for `pkg`, the result of `brew info --json=v1 \$pkg`.
+If `brew info` fails, throws an error.  If `brew info` returns an empty object,
+(e.g. "[]"), this returns an empty Dict.
+
+Note that running `brew info --json=v1` is somewhat expensive, so we cache the
+results in a global dictionary, and batching larger requests with this function
+similarly increases performance.
+"""
+function json(pkg::BrewPkg)
+    return json([pkg.name])[normalize_name(pkg.name)]
+end
+
+"""
+info(names::Vector{String})
+
+For each name in `names`, returns information about that particular package name
+as a BrewPkg.  This is our batched `String` -> `BrewPkg` converter.
+"""
+function info{T<:AbstractString}(names::Vector{T})
+    # Get the JSON representations of all of these packages
+    objs = json(names)
+
+    infos = BrewPkg[]
+    for name in names
+        obj = objs[normalize_name(name)]
+
+        # First, get name and version
+        obj_name = obj["name"]
+        version = make_version(obj_name, obj["versions"]["stable"])
+        version_str = obj["versions"]["stable"]
+
+        # Manually append the revision to the end of the version_str, as brew is wont to do
+        if obj["revision"] > 0
+            version_str *= "_$(obj["revision"])"
+        end
+        bottled = obj["versions"]["bottle"]
+
+        # If we actually have a keg, return whether it was poured
+        if !isempty(obj["installed"])
+            bottled = obj["installed"][1]["poured_from_bottle"]
+        end
+
+        # Push that BrewPkg onto our infos object
+        push!(infos, BrewPkg(obj_name, version, version_str, bottled))
+    end
+
+    # Return the list of infos
+    return infos
+end
+
+"""
+info(name::AbstractString)
+
+Returns information about a particular package name as a BrewPkg.  This is our
+basic `String` -> `BrewPkg` converter.
+"""
+function info(name::AbstractString)
+    return info([name])[1]
+end
+
+"""
+deps(name::AbstractString)
+
+Return a list of all direct dependencies of `name` as a `Vector{BrewPkg}`
+"""
+function deps(name::AbstractString)
+    obj = json(name)
+
+    # Iterate over all dependencies, removing optional dependencies
+    dependencies = String[dep for dep in obj["dependencies"]]
+    dependencies = filter(x -> !(x in obj["optional_dependencies"]), dependencies)
+    return info(dependencies)
+end
+
+"""
+deps(pkg::BrewPkg)
+
+Return a list of all direct dependencies of `pkg` as a `Vector{BrewPkg}`
+"""
+function deps(pkg::BrewPkg)
+    return deps(pkg.name)
+end
+
+"""
+deps_tree(name::AbstractString)
+
+Return a dictionary mapping every dependency (both direct and indirect) of `name`
+to a `Vector{String}` of all of its dependencies.  Used in `deps_sorted()`.
```

Comment:
Docstring and code don't agree, is it a `Dict{String,Vector{String}}` or a `Dict{String,Vector{BrewPkg}}` ?